### DISCUSSION TITLE
How to build query?

### DISCUSSION THREAD

**Discussion by vigith** [grafadruid/go-druid#4] [Q&A] [answered]
In the examples I see that the pattern used is to provide a byte string to `Load([]byte{})` and then `Execute`. Working with large strings could be tricky and might have to do templating for building it. Instead can we build a query (perhaps using the builder) and then pass it to `Execute`? If yes, could you please show me an example?
**Example**
The below is a `python` snippet, but I hope you understand what I am trying to ask.
```python
query.timeseries(
    datasource='twitterstream',
    granularity='day',
    intervals='2014-02-02/p4w',
    aggregations={'length': doublesum('tweet_length'), 'count': doublesum('count')},
    post_aggregations={'avg_tweet_length': (Field('length') / Field('count'))},
    filter=Dimension('first_hashtag') == 'sochi2014'
)
```

**Comment by jbguerraz** [comment, marked as answer]
Hello @vigith !

Thank you for your question :) 
I've just pushed an exampe for building structures https://github.com/grafadruid/go-druid/blob/master/examples/main.go
I've moved examples from parsing json to structures there https://github.com/grafadruid/go-druid/blob/master/examples/fromjson/main.go
Hope this helps :)

NB: You can chain methods calls like `f := filter.NewSelector().SetDimension("country").SetValue("France")` or you can go the one line per method call way 
```
f := filter.NewSelector()
f.SetDimension("country")
f.SetValue("France")
```

Enjoy!
